src/components/Router/index.js

```diff
@@ -232,14 +232,18 @@ class HistoryRouter extends Component {
     history.goBack()
   }
 
-  setStepIndex = (newStepIndex, newFlow) => {
+  setStepIndex = (newStepIndex, newFlow, excludeStepFromHistory) => {
     const {flow:currentFlow} = this.state
     const historyState = {
       step: newStepIndex,
       flow: newFlow || currentFlow,
     }
-    const path = `${location.pathname}${location.search}${location.hash}`
-    history.push(path, historyState)
+    if (excludeStepFromHistory) {
+      this.setState(historyState)
```

Comment:
is `this.props.onStepChange()` need to be called? I actually don't know what it does, just saw that  `onHistoryChange` calls it

Reply:
Yes, it does.

The component chosen for display is based on `this.state.flow` and `this.state.step`. So only when those change, the component will change.

Reply:
so is it being called? it seems that if we bypass the history that never gets called.

Reply:
It's being called if `excludeStepFromHistory` is set to `true`.

If we don't set `excludeStepFromHistory` (default behaviour), a new `history` entry will be created, and only then the state will be updated ([see here](https://github.com/onfido/onfido-sdk-ui/blob/2a93ba2384b17e1eb5ffc44019eadc05590fc214/src/components/Router/index.js#L189)).

If we do set `excludeStepFromHistory` to `true`, no history will be created but the state will be updated.

Reply:
Yep, I understood the behaviour from a code point of view. The question was whether we needed to call `onStepChange` if `excludeStepFromHistory==true`. I guess we don't have to.

Reply:
Oh... right, sorry I misread.

Actually I think you might have a point, instead of just `setState` we should do the same history does, without actually adding an entry, and that includes calling the `onStepChange` callback.

Reply:
who listens to `onStepChange`?
The PR has been merged, but you can create another PR to fix it, if it's indeed a problem.

Reply:
`CrossDeviceRouter`. If we go back on desktop, mobile should be notified of the step change